packages/uix-core/src/types.ts

```diff
@@ -164,6 +162,40 @@ export interface HostConnection<T = unknown> {
   invokeHostMethod: RemoteMethodInvoker<T>;
 }
 
+/**
+ * Guest UIs
+ * @internal
+ */
+export interface UIHostMethods {
+  onIframeResize(dimensions: DOMRect): void;
+}
+
+export type UIHostConnection<T = unknown> = HostConnection<T> & UIHostMethods;
+
+/** @public */
+export type GuestConnectionEvent<
+  Type extends string = string,
+  Detail = Record<string, unknown>
+> = NamedEvent<
+  Type,
+  Detail &
+    Record<string, unknown> & {
+      guestPort: GuestConnection;
+    }
+>;
+
+/** @public */
+export type GuestConnectionEvents<
+  HostApi extends Record<string, unknown> = Record<string, unknown>
+> =
+  | GuestConnectionEvent<"hostprovide">
+  | GuestConnectionEvent<"unload">
+  | GuestConnectionEvent<"beforecallhostmethod", HostMethodAddress<HostApi>>
+  | GuestConnectionEvent<
+      "iframeresize",
+      { dimensions: DOMRect; iframe: HTMLIFrameElement }
+    >;
+
```

Comment:
As part of the `GuestConnection` to `GuestEmitter` refactor, move the event definitions from the `Port` implementation to the `GuestEmitter` interface.